### ISSUE TITLE
TransactionScope Error

### ISSUE THREAD

**Issue by gregtayl** [neisbut/Npgsql.Bulk#19]
I'm getting an error when trying to use an ambient transaction. Does this library support TransactionScope or am I possibly misusing/misconfiguring something?

**Code:**
```
using (var scope = new TransactionScope(TransactionScopeAsyncFlowOption.Enabled))
using (var context = new DbContext())
{
    var uploader = new NpgsqlBulkUploader(context);
    uploader.Insert(entities, onConflict); 

    //...Other changes with the context
    
    await context.SaveChangesAsync().ConfigureAwait(false);

    scope.Complete();
}
```

**Error:**
System.InvalidOperationException: An ambient transaction has been detected. The ambient transaction needs to be completed before beginning a transaction on this connection.
   at Microsoft.EntityFrameworkCore.Storage.RelationalConnection.EnsureNoTransactions()
   at Microsoft.EntityFrameworkCore.Storage.RelationalConnection.BeginTransaction(IsolationLevel isolationLevel)
   at Microsoft.EntityFrameworkCore.RelationalDatabaseFacadeExtensions.<>c__DisplayClass18_0.<BeginTransaction>b__0(DatabaseFacade database)
   at Microsoft.EntityFrameworkCore.ExecutionStrategyExtensions.<>c__DisplayClass12_02.<Execute>b__0(DbContext c, TState s)
   at Npgsql.EntityFrameworkCore.PostgreSQL.Storage.Internal.NpgsqlExecutionStrategy.Execute[TState,TResult](TState state, Func3 operation, Func3 verifySucceeded) in C:\projects\npgsql-entityframeworkcore-postgresql\src\EFCore.PG\Storage\Internal\NpgsqlExecutionStrategy.cs:line 28
   at Microsoft.EntityFrameworkCore.ExecutionStrategyExtensions.Execute[TState,TResult](IExecutionStrategy strategy, Func2 operation, Func2 verifySucceeded, TState state)
   at Microsoft.EntityFrameworkCore.ExecutionStrategyExtensions.Execute[TState,TResult](IExecutionStrategy strategy, TState state, Func2 operation)
   at Microsoft.EntityFrameworkCore.RelationalDatabaseFacadeExtensions.BeginTransaction(DatabaseFacade databaseFacade, IsolationLevel isolationLevel)
   at Npgsql.Bulk.NpgsqlBulkUploader.Insert[T](IEnumerable1 entities, InsertConflictAction onConflict)

**Comment by neisbut** [comment]
Hi @gregtayl , I didn't test it with TransactionScope, but I think it should be possible to support it. I will check soon.

**Comment by neisbut** [comment]
Hi @gregtayl, I found tiny issue with TransactionScope handling. Fixed it in 0.7.0. Should work now.

**Comment by gregtayl** [comment]
Thank you!
